File: app/src/main/java/com/xq/phonecheck/identity/NumberRuleStore.kt
package com.xq.phonecheck.identity

import android.content.Context

object NumberRuleStore {
    private const val PREFS = "number_rules"
    private const val KEY_RULES = "rules_text"

    fun readText(context: Context): String {
        return context.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
            .getString(KEY_RULES, "") ?: ""
    }

    fun save(context: Context, text: String) {
        context.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
            .edit()
            .putString(KEY_RULES, text)
            .apply()
    }

    fun apply(context: Context, number: String): String {
        return NumberRules.apply(number, readText(context))
    }
}

File: ios/PhoneCheck/NumberRuleStore.swift
import Foundation

enum NumberRuleStore {
    private static let key = "number_rules"

    static func read() -> String {
        UserDefaults.standard.string(forKey: key) ?? ""
    }

    static func save(_ text: String) {
        UserDefaults.standard.set(text, forKey: key)
    }
}
